src/Evolution/DiscontinuousGalerkin/ExponentialFilter.hpp

```diff
@@ -0,0 +1,173 @@
+ *
+ * #### Design decision:
+ *
+ * - The reason for the `size_t` template parameter is to allow for different
+ * `Alpha` and `HalfPower` parameters for different tensors while still being
+ * able to cache the matrices.   If different `Alpha` or `HalfPower` parameters
+ * are desired for filtering different tensors, then multiple filters must be
+ * inserted into the ConstGlobalCache with different `FilterIndex` values. In
+ * the input file these will be specified as `ExpFilterFILTER_INDEX`, e.g.
+ * -  Filtering:
+ * -    ExpFilter0:
+ * -      Alpha: 12
+ * -      HalfPower: 32
+ * -    ExpFilter1:
+ * -      Alpha: 36
+ * -      HalfPower: 32
+ */
+template <size_t FilterIndex>
+class Exponential {
+ public:
+  /// \brief The value of `exp(-alpha)` is what the highest modal coefficient is
+  /// rescaled by.
+  struct Alpha {
+    using type = double;
+    static constexpr OptionString help =
+        "exp(-alpha) is rescaling of highest coefficient";
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  /*!
+   * \brief Half of the exponent in the exponential.
+   *
+   * \f{align*}{
+   *  c_i\to c_i \exp\left[-\alpha \left(\frac{i}{N}\right)^{2m}\right]
+   * \f}
+   */
+  struct HalfPower {
+    using type = unsigned;
+    static constexpr OptionString help =
+        "Half of the exponent in the generalized Gaussian";
+    static type lower_bound() noexcept { return 1; }
+  };
+
+  /// \brief Turn the filter off
+  ///
+  /// This option exists to temporarily disable the filter for debugging
+  /// purposes. For problems where filtering is not needed, the preferred
+  /// approach is to not compile the filter into the executable.
+  struct DisableForDebugging {
+    using type = bool;
+    static type default_value() noexcept { return false; }
+    static constexpr OptionString help = {"Disable the filter"};
+  };
+
+  using options = tmpl::list<Alpha, HalfPower, DisableForDebugging>;
+  static constexpr OptionString help = {"An exponential filter."};
+  static std::string name() noexcept {
+    return "ExpFilter" + std::to_string(FilterIndex);
+  }
+
+  Exponential() = default;
+
+  Exponential(double alpha, unsigned half_power,
+              bool disable_for_debugging) noexcept;
+
+  /// A cached matrix used to apply the filter to the given mesh
+  const Matrix& filter_matrix(const Mesh<1>& mesh) const noexcept;
+
+  bool disable_for_debugging() const noexcept { return disable_for_debugging_; }
+
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) noexcept;
+
+ private:
+  template <size_t LocalFilterIndex>
+  // NOLINTNEXTLINE(google-runtime-references)
+  friend bool operator==(const Exponential<LocalFilterIndex>& lhs,
+                         const Exponential<LocalFilterIndex>& rhs) noexcept;
+
+  double alpha_{36.0};
+  unsigned half_power_{16};
+  bool disable_for_debugging_{false};
+};
+
+template <size_t FilterIndex>
+Exponential<FilterIndex>::Exponential(const double alpha,
```

Comment:
If you are willing/interested, we could move the definitions to the cpp file and instantiate say 10 of the class? (0-9)  This will help combat compile times. I can do it later too